tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/SchemasSpec.scala

```diff
@@ -0,0 +1,142 @@
+import ch.epfl.bluebrain.nexus.tests.Identity.resources.Rick
+import ch.epfl.bluebrain.nexus.tests.iam.types.Permission.Organizations
+import io.circe.{Json, JsonObject}
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{Assertion, BeforeAndAfterAll}
+import org.scalatest.LoneElement._
+import org.scalatest.matchers.{HavePropertyMatchResult, HavePropertyMatcher}
+
+class SchemasSpec extends BaseSpec with EitherValuable with CirceEq with BeforeAndAfterAll {
+
+  private val orgId  = genId()
+  private val projId = genId()
+  private val id     = s"$orgId/$projId"
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    (for {
+      _ <- aclDsl.addPermission(
+             "/",
+             Rick,
+             Organizations.Create
+           )
+      _ <- adminDsl.createOrganization(orgId, orgId, Rick)
+      _ <- adminDsl.createProject(orgId, projId, kgDsl.projectJson(name = id), Rick)
+    } yield succeed).accepted
+    ()
+  }
+
+  private def minCount(expectedMinCount: Int) = HavePropertyMatcher[JsonObject, Option[Int]] { json =>
+    val actualMinCount = Json.fromJsonObject(json).hcursor.downField("minCount").as[Int]
+    HavePropertyMatchResult(
+      actualMinCount.contains(expectedMinCount),
+      "minCount",
+      Some(expectedMinCount),
+      actualMinCount.toOption
+    )
+  }
+
+  private def shapes(schema: Json): List[JsonObject] = {
+    schema.hcursor.downField("shapes").as[List[JsonObject]].getOrElse(Nil)
+  }
+
+  private def schemaPayloadWith(id: String, minCount: Int): Json = {
+    jsonContentOf("/kg/schemas/schema.json", "id" -> id, "minCount" -> minCount)
+  }
+
+  private def fetchTheSchema(schemaId: String) = { (resp: (Json, HttpResponse) => Assertion) =>
+    deltaClient.get[Json](s"/schemas/$id/${UrlUtils.encode(schemaId)}", Rick)(resp).accepted
+  }
+
+  "update a schema" in {
+    val schemaId = "updatable-schema"
+
+    thereIsASchema(schemaPayloadWith(schemaId, minCount = 1))
+
+    schemaIsUpdated(schemaId, 1, schemaPayloadWith(schemaId, minCount = 2))
+
+    fetchTheSchema(schemaId) { (json, response) =>
+      response.status shouldEqual StatusCodes.OK
+      shapes(json).loneElement should have(minCount(2))
+    }
+  }
+
+  private def payloadWithPowerLevelShape(id: String, maxPowerLevel: Int): Json = {
+    jsonContentOf("/kg/schemas/schema-with-power-level.json", "id" -> id, "maxPowerLevel" -> maxPowerLevel)
+  }
+
+  private def payloadThatImportsPowerLevelShape(id: String, importSchemaId: String): Json = {
+    jsonContentOf("/kg/schemas/schema-that-imports-power-level.json", "id" -> id, "import" -> importSchemaId)
+  }
+
+  private def thereIsASchema(payload: Json) = {
+    deltaClient
+      .post[Json](s"/schemas/$id", payload, Rick) { (_, response) =>
+        response.status shouldEqual StatusCodes.Created
+      }
+      .accepted
+  }
+
+  private def schemaIsUpdated(schemaId: String, revision: Int, payload: Json) = {
```

Comment:
`updateSchema` ?

Reply:
I don't know what would be the best way to rewrite those tests.
It seems a bit verbose but I don't have a strong opinion.
But it is one of the questions to ask ourselves when we will dump akka-http as the underlying http client